### ISSUE TITLE
Allow disabling warnings by comments

### ISSUE THREAD

**Issue by koalaman** [koalaman/shellcheck#43]
There should be a way to disable false positives in comments, e.g.

```
# shellcheck disable-msg SC2086
echo $1
```

These should be scoped for structure for which they appear, or the entire file if at the top.

**Comment by geoffmeyers** [comment]
feedback is that it works great.  maybe add it to the --help docs for ease of use?

**Comment by servel333** [comment]
It appears the directive will only apply to the line following the directive. For example:

    # shellcheck disable=SC2034
    AAAA=aaaa # < No 'appears unused'
    BBBB=bbbb # < Getting 'appears unused' error
    # shellcheck disable=SC2034
    CCCC=cccc # < No 'appears unused'
    DDDD=dddd # < Getting 'appears unused' error

I would expect the shellcheck directive to disable all of this error in the file, but if we need backwards compatibility, I would like a way to disable all of a specific type of error for the whole file.

My usecase: I have a `dotenv` file where I am declaring a lot of variables that I'm never using, because they are getting used in other files in my application. I would like to use `shellcheck` to syntax check the file but I don't need any 'appears unused' errors as they are irrelevant to my use.

**Comment by koalaman** [comment]
It's per structure, so you can do

```shellcheck disable=SC2034
{
AAAA=aaaa
BBBB=bbbb 
CCCC=cccc 
DDDD=dddd
}```

**Comment by servel333** [comment]
I used that method to solve my problem and it did work in my case, however I dislike the idea of having to change my code structure in order to make an ignore rule function. I would prefer to see a different directive to make all instances ignored.

Thanks for the feature though, it works great otherwise and this linter is very helpful for showing me some best-practices that I didn't know about.

**Comment by koalaman** [comment]
Thanks, glad you like it!

In the near future you're likely to see file-wide ignores, but right now they're per structure

**Comment by servel333** [comment]
Understood. Thanks.

**Comment by brother** [comment]
@servel333 also remember that you can disable it on the command line, might hide valid warnings in other files if you are processing more than one in a single go though. Just mentioning it as it looks like it's missing in this thread.


```bash
brother ~$ cat /tmp/test.sh
#!/bin/sh

echo "--no quote--"
for i in ${TMPDIR:=/tmp}; do echo "$i"; done
echo "------------"

echo "--- quote---"
for i in "${TMPDIR:=/tmp}"; do echo "$i"; done
echo "------------"

# shellcheck disable=SC2034
AAAA=aaaa # < No 'appears unused'
BBBB=bbbb # < Getting 'appears unused' error
# shellcheck disable=SC2034
CCCC=cccc # < No 'appears unused'
DDDD=dddd # < Getting 'appears unused' error
brother ~$ shellcheck /tmp/test.sh 

In /tmp/test.sh line 8:
for i in "${TMPDIR:=/tmp}"; do echo "$i"; done
         ^-- SC2066: Since you double quoted this, it will not word split, and the loop will only run once.


In /tmp/test.sh line 13:
BBBB=bbbb # < Getting 'appears unused' error
^-- SC2034: BBBB appears unused. Verify it or export it.


In /tmp/test.sh line 16:
DDDD=dddd # < Getting 'appears unused' error
^-- SC2034: DDDD appears unused. Verify it or export it.

brother ~$ shellcheck /tmp/test.sh -eSC2034

In /tmp/test.sh line 8:
for i in "${TMPDIR:=/tmp}"; do echo "$i"; done
         ^-- SC2066: Since you double quoted this, it will not word split, and the loop will only run once.

```

**Comment by servel333** [comment]
I didn't know you could disable it in the command line like that. I'm using Sublime that's auto-running the command though and so for my use case it doesn't solve my problem. Good to know though, thanks.

**Comment by koalaman** [comment]
If Sublime isn't configurable you can set SHELLCHECK_OPTS in recent versions of shellcheck

**Comment by servel333** [comment]
I actually don't want to disable this warning for all files I'm checking, just specific ones.
